src/lib.rs

```diff
@@ -556,6 +616,7 @@ violation information
 https://www.w3.org/TR/CSP/#violation
 */
 #[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
```

Comment:
Required so that we can send violations via IPC calls to the relevant `FetchTaskListener`. The other enums/structs already have this, so I think it is fine to add?

Reply:
Yes, this is fine.